config/initializers/availability_checker.rb

```diff
@@ -0,0 +1,7 @@
+# Instantiate the Source Availability Checker Singleton
+Rails.application.config.after_initialize do
+  if defined?(::Rails::Server)
+    SourceAvailabilityChecker.instance
```

Comment:
Ehh idk about a thread in the API server, if we scale this up we'll get one of these per pod.  I pictured the actual scheduling of the availability check as a separate worker.  @Fryguy ?

Reply:
makes sense. we need one and only one of those. 👍

Reply:
Marking all as WIP for now. One other potential issue bugging me this morning is with
https://github.com/ManageIQ/topological_inventory-orchestrator/pull/32 coming in the picture.
Let's say we resolve the one and only one instance (worker, etc), which is needed. Next, sources being unavailable, the worker sees that and sends a message to the operation worker side of things, he does his magic and sees a source up and reachable, then he sends a request to update the source's availability_status, the patch might fail because there may not be a source running to take that api call because the orchestrator didn't bring one up (availability_status being down).

Reply:
Ok, re: ^ I think we're good as the orchestrator only brings up/down collectors, so the sources project and the operations workers would always be running.

Reply:
@abellotti yeah the operations workers are per type not per source :+1: